framework/py/flwr/cli/hub_app/publish.py

```diff
@@ -0,0 +1,332 @@
+
+
+def _validate_credentials_content(creds_path: Path) -> dict[str, str]:
+    """Load and validate the credentials file content.
+
+    Ensures required keys exist:
+      - AUTHN_TYPE_JSON_KEY
+      - ACCESS_TOKEN_KEY
+      - REFRESH_TOKEN_KEY
+    """
+    try:
+        content = json.loads(creds_path.read_text(encoding="utf-8"))
+    except (OSError, json.JSONDecodeError) as err:
+        typer.secho(
+            f"Invalid credentials file at '{creds_path}': {err}",
+            fg=typer.colors.RED,
+            err=True,
+        )
+        raise typer.Exit(code=1) from err
+
+    required_keys = [AUTHN_TYPE_JSON_KEY, ACCESS_TOKEN_KEY, REFRESH_TOKEN_KEY]
+    missing = [key for key in required_keys if key not in content]
+
+    if missing:
+        typer.secho(
+            f"Credentials file '{creds_path}' is missing "
+            f"required key(s): {', '.join(missing)}.",
+            fg=typer.colors.RED,
+            err=True,
+        )
+        raise typer.Exit(code=1)
+
+    return content  # type: ignore[no-any-return]
+
+
+def publish(
+    app: Annotated[
+        Path,
+        typer.Argument(
+            help="Project directory to upload (defaults to current directory)."
+        ),
+    ] = Path("."),
+    federation: Annotated[
+        Optional[str],
+        typer.Argument(help="Name of the federation used for app publishing."),
+    ] = None,
+    token: Annotated[
+        Optional[str],
+        typer.Option(
+            "--token",
+            help="Bearer token for Platform API.",
+        ),
+    ] = None,
+) -> None:
+    """Upload all project files to the Platform API using multipart/form-data."""
+    # Check the credentials path
+    if not token:
+        if not federation:
+            typer.secho(
+                "❌ Please specify the federation used for "
+                "login before publishing app.",
+                fg=typer.colors.RED,
+                err=True,
+            )
+            raise typer.Exit(code=1)
+
+        creds_path = app.absolute() / FLWR_DIR / CREDENTIALS_DIR / f"{federation}.json"
+        if not creds_path.is_file():
+            typer.secho(
+                "❌ Please login before publishing app.",
+                fg=typer.colors.RED,
+                err=True,
+            )
+            raise typer.Exit(code=1)
+
+        # Load and validate credentials
+        creds = _validate_credentials_content(creds_path)
+        token = creds[ACCESS_TOKEN_KEY]
+
+    # Collect & validate app files
+    files = _collect_files(app)
+    _validate_files(files)
+
+    # Build and POST multipart
+    with ExitStack() as stack:
+        files_param = _build_multipart_files_param(files, stack)
+        try:
+            resp = _post_files(files_param, token)
+        except requests.RequestException as err:
+            typer.secho(f"❌ Network error: {err}", fg=typer.colors.RED, err=True)
+            raise typer.Exit(code=1) from err
+
+    if resp.status_code // 100 == 2:
```

Comment:
This looks a bit strange.  Would it be better to check for a specific status code instead of doing `// 100`?